code/scripting/api/objs/debriefing.cpp

```diff
@@ -0,0 +1,115 @@
+#include "debriefing.h"
+
+namespace scripting {
+namespace api {
+
+//**********HANDLE: debriefing
+ADE_OBJ(l_DebriefStage, debrief_stage, "debriefing_stage", "Debriefing stage handle");
+
+ADE_VIRTVAR(Text, l_DebriefStage, nullptr, "The text of the stage", "debriefing_stage", "The text")
+{
+	debrief_stage* stage = nullptr;
+	if (!ade_get_args(L, "o", l_DebriefStage.GetPtr(&stage))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", stage->text);
+}
+
+ADE_VIRTVAR(AudioFilename,
+	l_DebriefStage,
+	nullptr,
+	"The filename of the audio file to play",
+	"debriefing_stage",
+	"The file name")
+{
+	debrief_stage* stage = nullptr;
+	if (!ade_get_args(L, "o", l_DebriefStage.GetPtr(&stage))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", stage->voice);
+}
+
+ADE_VIRTVAR(Recommendation, l_DebriefStage, nullptr, "The recommendation text of the stage", "debriefing_stage", "The recommendation text")
+{
+	debrief_stage* stage = nullptr;
+	if (!ade_get_args(L, "o", l_DebriefStage.GetPtr(&stage))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", stage->recommendation_text);
+}
+
+ADE_VIRTVAR(isValid,
```

Comment:
`isValid` is used, by convention, to see whether the Lua object itself is a valid object.  This should have another name, perhaps `isVisible`.